### DISCUSSION TITLE
With `display` (or other IM commands), how to get the pixel coordinates of mouse clicks logged?

### DISCUSSION THREAD

**Discussion by emk2203** [ImageMagick/ImageMagick#2880] [Help] [answered]
I am looking for an option to log the pixel coordinates when I click with the mouse in a displayed picture. Ideally with RGB values.

When I middle-click into a displayed image, I get the coordinates and the color in the magnify subwindow, so they are already found. But how can I get this either into a log or to stdout? I already tried to use `-debug`, but couldn't find anything relevant.

Using IM 6.9.10.23 at the moment, but willing to go to 7 if necessary.

**Comment by fmw42** [comment, marked as answer]
As far as I know, ImageMagick currently cannot do that. Its GUI on Unix-like systems is rather primitive. You could try the Pointpicker extension for ImageJ. I use that from time to time.

**Reply by emk2203** [reply]
Thanks, I suspected as much and already downloaded ImageJ, just wanted to wrap up my search. I assume when this statement comes from you, it's more or less the authoritative answer for IM.

A bit strange that this functionality is not more in demand. Getting pixels of interest in an image for future script-based processing shouldn't be this exotic, or is it?

**Reply by fmw42** [reply]
I am not the final authority. That would have to come from the IM developers. But ImageMagick is primarily a command line tool. The GUI is very primitive.
